src/soter/openssl/soter_rsa_key_pair_gen.c

```diff
@@ -44,10 +44,24 @@ static int rsa_key_length(unsigned size)
 
 soter_rsa_key_pair_gen_t* soter_rsa_key_pair_gen_create(const unsigned key_length)
 {
-    SOTER_CHECK_PARAM_(rsa_key_length(key_length) > 0);
```

Comment:
why you decided to expand all macros? we have some problems with them in the context of openssl3.0 support?

Reply:
> why you decided to expand all macros? we have some problems with them in the context of openssl3.0 support?

This is not related to OpenSSL 3.0 directly. I just have a problem with untangling the control flow here. Remembering which of those test for condition to be true and which test for it to be false, which return SOTER_FAIL and which return NULL, and which do something else, etc.

While these macros *do* have a benefit of logging failed conditions in debug builds, which (unsurprisingly) helps with debugging, it's not *that* important. In most cases we'd like to debug log OpenSSL failures too, since *those* are actually hard to understand. However, almost none of OpenSSL calls are instrumented with these macros.

I hold a belief that C macros affecting control flow are pure evil and should be abolished. If logging is necessary, I'd rather write
```c
if (rsa_key_length(key_length) <= 0) {
    // Prints to stderr in debug builds, does nothing in normal builds:
    SOTER_DEBUG("invalid key_length %d (%d bits)", key_length, rsa_key_length(key_length));
    return SOTER_INVALID_PARAMETER;
}
```
which surely is more verbose, but the control flow is much more clear too.